File: common_business/src/main/java/com/xiaoyv/busines/room/dao/FtpDao.java
package com.xiaoyv.busines.room.dao;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import com.xiaoyv.busines.room.entity.FtpEntity;

import java.util.List;

/**
 * ftpDao
 *
 * @author why
 * @since 2020/11/29
 **/
@Dao
public interface FtpDao {
    @Query("SELECT * FROM ftp")
    List<FtpEntity> getAll();

    @Query("SELECT * FROM ftp WHERE ip IN (:ip)")
    List<FtpEntity> loadAllByIps(int... ip);

    @Query("SELECT * FROM ftp WHERE label LIKE :label")
    List<FtpEntity> findAllByLabel(String label);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertAll(FtpEntity... ftpEntities);

    @Update
    void update(FtpEntity... ftpEntities);

    @Delete
    void delete(FtpEntity... FtpEntity);
}

File: common_business/src/main/java/com/xiaoyv/busines/room/dao/FtpDao.kt
package com.xiaoyv.busines.room.dao

import androidx.room.*
import com.xiaoyv.busines.room.entity.FtpEntity

/**
 * ftpDao
 *
 * @author why
 * @since 2020/11/29
 */
@Dao
interface FtpDao {
    @Query("SELECT * FROM ftp")
    fun queryAll(): List<FtpEntity>

    @Query("SELECT * FROM ftp WHERE id =:id LIMIT 1")
    fun queryById(id: Long): FtpEntity?

    @Query("SELECT * FROM ftp WHERE ip IN (:ip)")
    fun queryByIps(vararg ip: Int): List<FtpEntity>

    @Query("SELECT * FROM ftp WHERE label LIKE :label")
    fun queryLabel(label: String): List<FtpEntity>

    @Query("SELECT DISTINCT `group` FROM ftp ORDER BY `group` COLLATE LOCALIZED ASC")
    fun queryGroup(): List<String>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(vararg ftpEntities: FtpEntity)

    @Update
    fun update(vararg ftpEntities: FtpEntity)

    @Delete
    fun delete(vararg FtpEntity: FtpEntity)
}